070-Integrations/100-oauth.mdx

```diff
@@ -0,0 +1,172 @@
+
+| Parameter       | Description                                                                                                                                                                              |
+| --------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `client_id`     | The client ID of the application.                                                                                                                                                        |
+| `redirect_uri`  | The URL that Xata should redirect the user back to after the user signs in. This URL must be registered with the application.                                                            |
+| `response_type` | Must be set to `code`.                                                                                                                                                                   |
+| `scope`         | Scopes that the application is requesting. Can be repeated to request multiple scopes. Right now, the only supported scope is `admin:all` and can be ignored as it is the default scope. |
+| `state`         | A string that the application can use to maintain state between the request and the callback. Also used to prevent CSRF attacks.                                                         |
+
+For example:
+
+```
+https://app.xata.io/integrations/oauth/authorize?client_id=123&redirect_uri=https://example.com/callback&response_type=code&scope=admin:all&state=abc123
+```
+
+### Redirecting the User Back to the Application
+
+After the user signs in, Xata will redirect the user back to the application. Xata will include the following query parameters:
+
+| Parameter    | Description                                                          |
+| ------------ | -------------------------------------------------------------------- |
+| `code`       | The authorization code.                                              |
+| `state`      | The state that was provided in the original request.                 |
+| `color-mode` | The color mode of the user's Xata account. Can be `light` or `dark`. |
+
+For example:
+
+```
+https://example.com/callback?code=abc123&state=abc123&color-mode=light
+```
+
+### Exchanging the Authorization Code for an Access Token
+
+To exchange the authorization code for an access token, the application must make a `POST` request to the following URL:
+
+```
+https://app.xata.io/api/integrations/oauth/token
+```
+
+The application must include the following parameters in the request body:
+
+| Parameter       | Description                           |
+| --------------- | ------------------------------------- |
+| `client_id`     | The client ID of the application.     |
+| `client_secret` | The client secret of the application. |
```

Comment:
Should we give some context as to how the client id/secret will be provided (i.e. provided by us after we register an application)?

Reply:
Don't think so, let's wait for them to connect via email